sdk/src/main/java/io/opentelemetry/sdk/trace/SpanInterceptor.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.trace;
+
+import io.opentelemetry.trace.Span;
+
+/**
+ * SpanInterceptor is the interface {@code TracerSdk} uses to allow synchronous hooks for when a
+ * {@code Span} is started or when a {@code Span} is ended.
+ */
+public interface SpanInterceptor {
```

Comment:
`Interceptor` is confusing name. Sounds like something that intercepts that standard flow. When in reality intercept IS a standard flow. Perhaps `Processor`, `Channel`, `Pipeline` instead?

Reply:
`Reporter`?

Reply:
Reporter semantically sounds the same as exporter. This class is more like a channel or pipeline where spans can even be filtered out. I'm OK taking PR as is and discuss name later.